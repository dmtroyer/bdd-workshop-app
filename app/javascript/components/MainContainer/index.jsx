import React from 'react';
import PropTypes from 'prop-types';

const MainContainer = ({ children }) => <main className="m-5">{children}</main>;
MainContainer.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default MainContainer;
