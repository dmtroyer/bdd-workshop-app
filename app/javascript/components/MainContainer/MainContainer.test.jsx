import React from 'react';
import { shallow } from 'enzyme';
import MainContainer from '.';

describe('MainContainer', () => {
  it('renders correctly for 1 child', () => {
    const wrapper = shallow(
      <MainContainer>
        <span>single child</span>
      </MainContainer>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly for many children', () => {
    const wrapper = shallow(
      <MainContainer>
        <div>one</div>
        <div>two</div>
      </MainContainer>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
