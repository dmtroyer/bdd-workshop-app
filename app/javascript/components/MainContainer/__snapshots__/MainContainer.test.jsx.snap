// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer renders correctly for 1 child 1`] = `
<main
  className="m-5"
>
  <span>
    single child
  </span>
</main>
`;

exports[`MainContainer renders correctly for many children 1`] = `
<main
  className="m-5"
>
  <div>
    one
  </div>
  <div>
    two
  </div>
</main>
`;
